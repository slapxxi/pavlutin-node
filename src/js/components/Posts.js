import { chunk } from 'lodash';
import React from 'react';
import Transition from 'react-transition-group/CSSTransitionGroup';
import PostPreview from './PostPreview';

function Posts({ posts, tag }) {
  if (posts.length === 0) {
    return <section className="posts"><p>No posts found.</p></section>;
  }
  return (
    <section className="posts">
      <ul className="nav-list">
        <Transition
          transitionName="fade"
          transitionEnterTimeout={250}
          transitionLeaveTimeout={250}
        >
          {
          chunk(posts, 3).map(ch => (
            <div className="posts__row">
              { ch.map(p => (
                <li key={p.id} className="posts__post">
                  <PostPreview post={p} activeTag={tag} />
                </li>
              ))}
            </div>
          ))
          }
        </Transition>
      </ul>
    </section>
  );
}

export default Posts;
