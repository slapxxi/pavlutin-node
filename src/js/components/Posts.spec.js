import React from 'react';
import render from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Posts from './Posts';

const posts = [
  { id: 0, title: 'First Post', tags: ['js'] },
  { id: 1, title: 'Second Post' },
];

it('renders posts', () => {
  const tree = render.create(<Router><Posts posts={posts} /></Router>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders posts with active tag', () => {
  const tree = render.create(
    <Router>
      <Posts posts={posts} tag="tag" />
    </Router>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders message when there are no posts', () => {
  const tree = render.create(<Posts posts={[]} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
