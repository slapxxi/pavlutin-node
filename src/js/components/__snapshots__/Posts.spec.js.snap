// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders message when there are not posts 1`] = `
<p>
  No posts found.
</p>
`;

exports[`renders posts 1`] = `
<section
  className="posts"
>
  <ul
    className="nav-list"
  >
    <div
      className="posts__row"
    >
      <li
        className="posts__post"
      >
        <div
          className="preview"
        >
          <header
            className="preview__header"
          >
            <h2
              className="preview__title"
            >
              <a
                href="/blog/undefined"
                onClick={[Function]}
                title="Read First Post"
              >
                First Post
              </a>
            </h2>
            <div
              className="preview__meta"
            >
              <span
                aria-hidden={true}
                className="fa fa-clock-o"
              />
               
               
              a few seconds ago
            </div>
            <ul
              className="tags preview__tags"
            >
              <li
                className="tags__tag"
              >
                <a
                  href="/blog/tag/js"
                  onClick={[Function]}
                >
                  js
                </a>
              </li>
            </ul>
          </header>
          <div
            className="preview__description"
          >
            <div>
              <p>
                There's no description.
              </p>
            </div>
          </div>
          <div
            className="preview__button"
          >
            <a
              href="/blog/undefined"
              onClick={[Function]}
            >
              Keep Reading
            </a>
          </div>
        </div>
      </li>
      <li
        className="posts__post"
      >
        <div
          className="preview"
        >
          <header
            className="preview__header"
          >
            <h2
              className="preview__title"
            >
              <a
                href="/blog/undefined"
                onClick={[Function]}
                title="Read Second Post"
              >
                Second Post
              </a>
            </h2>
            <div
              className="preview__meta"
            >
              <span
                aria-hidden={true}
                className="fa fa-clock-o"
              />
               
               
              a few seconds ago
            </div>
          </header>
          <div
            className="preview__description"
          >
            <div>
              <p>
                There's no description.
              </p>
            </div>
          </div>
          <div
            className="preview__button"
          >
            <a
              href="/blog/undefined"
              onClick={[Function]}
            >
              Keep Reading
            </a>
          </div>
        </div>
      </li>
    </div>
  </ul>
</section>
`;

exports[`renders posts with active tag 1`] = `
<section
  className="posts"
>
  <ul
    className="nav-list"
  >
    <div
      className="posts__row"
    >
      <li
        className="posts__post"
      >
        <div
          className="preview"
        >
          <header
            className="preview__header"
          >
            <h2
              className="preview__title"
            >
              <a
                href="/blog/undefined"
                onClick={[Function]}
                title="Read First Post"
              >
                First Post
              </a>
            </h2>
            <div
              className="preview__meta"
            >
              <span
                aria-hidden={true}
                className="fa fa-clock-o"
              />
               
               
              a few seconds ago
            </div>
            <ul
              className="tags preview__tags"
            >
              <li
                className="tags__tag"
              >
                <a
                  href="/blog/tag/js"
                  onClick={[Function]}
                >
                  js
                </a>
              </li>
            </ul>
          </header>
          <div
            className="preview__description"
          >
            <div>
              <p>
                There's no description.
              </p>
            </div>
          </div>
          <div
            className="preview__button"
          >
            <a
              href="/blog/undefined"
              onClick={[Function]}
            >
              Keep Reading
            </a>
          </div>
        </div>
      </li>
      <li
        className="posts__post"
      >
        <div
          className="preview"
        >
          <header
            className="preview__header"
          >
            <h2
              className="preview__title"
            >
              <a
                href="/blog/undefined"
                onClick={[Function]}
                title="Read Second Post"
              >
                Second Post
              </a>
            </h2>
            <div
              className="preview__meta"
            >
              <span
                aria-hidden={true}
                className="fa fa-clock-o"
              />
               
               
              a few seconds ago
            </div>
          </header>
          <div
            className="preview__description"
          >
            <div>
              <p>
                There's no description.
              </p>
            </div>
          </div>
          <div
            className="preview__button"
          >
            <a
              href="/blog/undefined"
              onClick={[Function]}
            >
              Keep Reading
            </a>
          </div>
        </div>
      </li>
    </div>
  </ul>
</section>
`;
